src/include/add-on/add-on-workflow.rb

```diff
@@ -878,6 +878,9 @@ def RemoveSelectedAddOn(selected)
         Pkg.SourceDelete(media)
       end
 
+      # Deselect product from libzypp
+      Pkg.ResolvableNeutral(AddOnProduct.add_on_products[selected]["product"], :product, true)
```

Comment:
I think the last parameter should be `false`, see https://github.com/yast/yast-pkg-bindings/blob/78f5ce710aa8973f50227150ed570467d0d71885/src/Resolvable_Install.cc#L400.

Forcing the change on the user level is a probably not needed in this case as the product was originally selected by YaST, not by user.

Could you test if it works with `false`?